### ISSUE TITLE
Can Bash on Ubuntu For Windows be used with the VSCode extension?

### ISSUE THREAD

**Issue by ghost** [microsoft/vscode-cpptools#85]
Can Bash on Ubuntu for Windows be used in place of Cygwin by the VSCode C++ extension for debugging?

**Comment by sridmad** [comment]
@rogersachan, Can you give us more context?

**Comment by ghost** [comment]
Can Bash on Ubuntu for Windows be used in place of Cygwin for the VSCode C++ extension?

**Comment by sridmad** [comment]
@delmyers do you know the answer?

**Comment by ghost** [comment]
@delmyers any update?

**Comment by pieandcakes** [comment]
@rogersachan Currently we only support MinGW and Cygwin. We can take this as a feature request

**Comment by KHeresy** [comment]
Please support this!
This make VSCode easier to use in Windows 10

**Comment by spookysys** [comment]
++ on this!
This would make using windows for developing cross-platform software a lot more feasible

**Comment by ghost** [comment]
The Anniversary Update is now out! This at least gives you all some reason
to support Bash for Windows, no?

**Comment by rrciisc** [comment]
+1 for this.

**Comment by arctanx0** [comment]
+1, please

**Comment by delmyers** [comment]
Thanks everyone for the feedback. There is currently a limitation in Windows that is preventing us from doing this correctly. We are waiting on the Windows team to finish off their work.

**Comment by ghost** [comment]
@delmyers Will the new Windows 10 Creators Update allow this?

**Comment by mzvast** [comment]
Windows team recently introduced improvements in symbolic link creation on windows, will that help with this?
https://blogs.windows.com/buildingapps/2016/12/02/symlinks-windows-10/

**Comment by incomingstick** [comment]
Is there any update on this feature? We haven't heard anything for a few months now.

It would be AWESOME if we could use the lxss binaries in VSCode (I would love to use its git, cmake, etc.)

**Comment by pieandcakes** [comment]
@incomingstick I have made a prototype to try this scenario and we are waiting on some bugfixes in the `Bash on Windows` side. I am waiting on a response as to when those will be released in order for our scenario to work.

**Comment by pieandcakes** [comment]
Update: We are waiting for necessary fixes from the WSL team that should be out in the next major update to Windows 10 called Creator's Update. We also have some bug fixes in our extension that should be released with our monthly February release. This will enable users to use the `pipeTransport` option to connect to bash on WSL.

**Comment by chris-ray** [comment]
+1 please :-)

**Comment by incomingstick** [comment]
Will there be some sort of documentation for this feature, and an announcement for when this will be available?

**Comment by pieandcakes** [comment]
@incomingstick I will add a documentation page in the repo on how to get it to work once Windows 10 Creator's Update is available to the general public.

**Comment by nayato** [comment]
@pieandcakes with Creators Update out now, is there an update on this?

**Comment by pieandcakes** [comment]
@nayato I'm working on the instructions. The official release is April 11th and I'm targeting to have it ready in time.

**Comment by timsneath** [comment]
@pieandcakes It would be helpful if the extension supports using environment variables in the `c_cpp_properties.json` for include files. I'd like to be able to write something like this to reference include files:

```
"includePath": [
   "%localappdata%/lxss/rootfs/usr/include",
   "%localappdata%/lxss/rootfs/usr/local/include"            
],
```

Today, this doesn't work and I have to replace `%localappdata%` with `c:/users/tims/appdata/local`, which is obviously not portable across multiple user accounts. 

Let me know if this isn't currently on your radar, and I'd be happy to open up a separate issue.

**Comment by bobbrow** [comment]
@timsneath, have you tried using `${localappdata}` instead of `%localappdata%`?  I just did a quick test and it works for me.  If not, can you please open a new issue?

**Comment by timsneath** [comment]
Ah, brilliant! This worked a treat. Thanks so much, @bobbrow - I'm learning new stuff here!

**Comment by pieandcakes** [comment]
@nayato and anyone else: I have a PR out for the documentation change. If you would like to preview it: https://github.com/Microsoft/vscode-cpptools/pull/626

Thanks!

**Comment by pieandcakes** [comment]
I have updated our documentation to include how to debug on Bash on Windows. With the Creators Update being release on 4/11/17, this extension will now be able to debug those applications. For more information, the documentation page can be found here: [Windows Subsystem for Linux](https://github.com/Microsoft/vscode-cpptools/blob/master/Documentation/Debugger/gdb/Windows%20Subsystem%20for%20Linux.md)

**Comment by rrciisc** [comment]
@pieandcakes thanks for putting this in documentation . I have tested launch configuration and its working perfect.

It will be great if you can put WSL make task setup as well (tasks.json file).
Thanks in advance.
